ex2_hal/athena/equipment_handler/source/housekeeping_athena.c

```diff
@@ -90,31 +85,69 @@ uint8_t Athena_get_solar_supply_curr() {
  * 		Last found error will be returned. else no error returned
  */
 int Athena_getHK(athena_housekeeping *athena_hk) {
-    int temporary;
+    int temp_status;
     int return_code = 0;
 
     /*Add athena HAL housekeeping getters here and put fields in h file
     create HAL functions here following format of existing
     also add endianness conversion in Athena_hk_convert_endianness*/
-    temporary = HAL_get_temp_all(athena_hk->temparray);
+    temp_status = HAL_get_temp_all(&athena_hk->MCU_core_temp, &athena_hk->converter_temp);
+
+    // Get last 8 digits of the software version
+    memcpy(athena_hk->OBC_software_ver, ex2_hk_version, 8 * sizeof(char));
 
     // Get OBC uptime: Seconds = value*10. Max = 655360 seconds (7.6 days)
     athena_hk->OBC_uptime = Athena_get_OBC_uptime();
 
+    // Get boot info
+    boot_info info;
+    if (eeprom_get_boot_info(&info)) {
+        return_code = -1;
+    }
+    athena_hk->boot_cnt = info.count;
+    athena_hk->last_reset_reason = info.reason.swr_reason;
+    athena_hk->boot_src = info.reason.rstsrc;
+
     // Get solar panel supply current
     athena_hk->solar_panel_supply_curr = Athena_get_solar_supply_curr();
 
-    // placeholder for software version
-    athena_hk->OBC_software_ver = software_version;
-
-    if (temporary != 0)
-        return_code = temporary;
+    if (temp_status != 0)
+        return_code = temp_status;
 
     // Get solar panel supply current
     athena_hk->solar_panel_supply_curr = Athena_get_solar_supply_curr();
 
-    if (temporary != 0)
-        return_code = temporary;
+    // Get SD card usage percentages
+#if defined(HAS_SD_CARD)
+    int32_t iErr;
+    REDSTATFS volstat;
+    iErr = red_statvfs("VOL0:", &volstat);
+    athena_hk->vol0_usage_percent =
+        (uint8_t)((float)(volstat.f_bfree) * 100.0 /
+                  ((float)(volstat.f_blocks))); // assuming block size == sector size = 512B
+    if (iErr == -1) {
+        exit(red_errno);
+    }
+
+#ifdef IS_ATHENA_V2
+    iErr = red_statvfs("VOL1:", &volstat);
+    athena_hk->vol1_usage_percent =
+        (uint8_t)((float)(volstat.f_bfree) * 100.0 /
+                  ((float)(volstat.f_blocks))); // assuming block size == sector size = 512B
+    if (iErr == -1) {
+        exit(red_errno);
+    }
+
+#else  // IS_ATHENA_V2
+    athena_hk->vol1_usage_percent = 0; // stub if no card
```

Comment:
```suggestion
    athena_hk->vol1_usage_percent = 255; // stub if no card
```

Reply:
This makes it clearer that there is no SD card as opposed to an SD card that is 0% full. Do the same for line 146